zebrad/tests/acceptance.rs

```diff
@@ -1689,6 +1755,33 @@ fn zebra_tracing_conflict() -> Result<()> {
     Ok(())
 }
 
+/// Start 2 zebrad nodes using the same RPC listener port, but different
+/// state directories and Zcash listener ports. The first node should get
+/// exclusive use of the port. The second node will panic.
+#[test]
+fn zebra_rpc_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created RPC listen_addr
+    let mut config = default_test_config()?;
+    config.rpc.listen_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = testdir()?.with_config(&mut config)?;
+    let regex1 = regex::escape(&format!(r"Opened RPC endpoint at {}", listen_addr));
+
+    // From another folder create a configuration with the same endpoint.
+    // `rpc.listen_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = testdir()?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, "Unable to start RPC server")?;
```

Comment:
It seems like this test is failing because the RPC port isn't quite open when the second process launches.

We could try putting a short sleep after it logs the RPC port message?
https://github.com/ZcashFoundation/zebra/blob/397ba1fef73addc5a543a164daa1ce56bdcdeac1/zebrad/tests/acceptance.rs#L1761-L1762

Reply:
I am not sure how much will be enough here as this test pass locally. I added 2 seconds to test the CI in https://github.com/ZcashFoundation/zebra/pull/3641/commits/b46d6c64c9d4cd7e078421e90588b0bb8fa7defe

I don't want to make this delay too big if possible as there are several tests using the function. Lets see if it helps.

Reply:
2 seconds fixed the first error.

Now we have a different error because the second node is slow:
> conflicted node2 was still running, but the test expected a panic

Which is happening because this delay is too short:
https://github.com/ZcashFoundation/zebra/blob/4fc10e52570721c41f2a9f1bced2338449dd6450/zebrad/tests/acceptance.rs#L1771-L1773

I suggest changing `LAUNCH_DELAY` to about 15 seconds.

The tests are failing in CI because one of the DNS seeders is failing, which takes 10 seconds to retry. (The calculation is `DNS_LOOKUP_TIMEOUT * (MAX_SINGLE_PEER_RETRIES + 1)`.)

Then after that Zebra needs time to connect to initial peers, get addresses, launch the other services, and launch RPC.

You could also move the RPC open right after the mempool is created? That might speed things up a bit.

Reply:
Another alternative would be to wait until the second Zebra instance logs that it is trying to open the conflicting port, then wait a few more seconds.

But that might mean changing all the tests to take a "trying" log regex.

Reply:
(This seems to be fixed now.)